components/lang/src/main/java/io/cellery/impl/CreateInstance.java

```
@@ -81,32 +124,144 @@
+            Meta rootMeta = (Meta) dependencyTree.getRoot().getData();
+            BMap<String, BValue> rootCellInfo = new BMap<>(new BArrayType(BTypes.typeString));
+            rootCellInfo.put("org", new BString(rootMeta.getOrg()));
+            rootCellInfo.put("name", new BString(rootMeta.getName()));
+            rootCellInfo.put("ver", new BString(rootMeta.getVer()));
+            rootCellInfo.put("instanceName", new BString(rootMeta.getInstanceName()));
+            bmap = BLangConnectorSPIUtil.createBStruct(ctx,
+                    CelleryConstants.CELLERY_PACKAGE,
+                    CelleryConstants.INSTANCE_STATE_DEFINITION,
+                    rootCellInfo, isInstanceRunning(rootMeta.getInstanceName(), rootMeta.getKind()));
+
+            bValueArray.add(runCount.getAndIncrement(), bmap);
+            dependencyInfo = generateDependencyInfo();
+
+            if (startDependencies) {
+                try {
+                    // Display the dependency tree info table
+                    displayDependentCellTable();
+                    dependencyInfo.forEach((alias, info) -> {
+                        String depInstanceName = ((BString) ((BMap) info).getMap().get(INSTANCE_NAME)).stringValue();
+                        String depKind = ((BString) ((BMap) info).getMap().get(KIND)).stringValue();
+                        bmap = BLangConnectorSPIUtil.createBStruct(ctx,
+                                CelleryConstants.CELLERY_PACKAGE,
+                                CelleryConstants.INSTANCE_STATE_DEFINITION,
+                                info, isInstanceRunning(depInstanceName, depKind), alias);
+                        bValueArray.add(runCount.getAndIncrement(), bmap);
+                    });
+                    // Start the dependency tree
+                    startDependencyTree(dependencyTree.getRoot());
+                } catch (Exception e) {
+                    printWarning("Unable to start dependencies. " + e);
```

Shouldn't we propagate this error?

---

Fixed in all places.